### PR TITLE
Added alias support to MLflow client

### PR DESCRIPTION
**Pull request by arpitjasa-db**
## Related Issues/PRs

https://github.com/mlflow/mlflow/pull/8001
https://github.com/mlflow/mlflow/pull/8055

## What changes are proposed in this pull request?

Add aliases support to the MLflow Python client and add tests.

## How is this patch tested?

- [X] Existing unit/integration tests
- [X] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Add aliases support to the MLflow Python client and add tests.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/tracking/client.py [hunk 1]
```diff
@@ -2904,3 +2909,257 @@ def print_model_version_info(mv):
...
+            model_uri = "runs:/{}/sklearn-model".format(run.info.run_id)
+            mv = client.create_model_version(name, model_uri, run.info.run_id)
+            print_model_version_info(mv)
+
+            # Set registered model alias
+            client.set_registered_model_alias(name, "test-alias", mv.version)
+            print("\n")
+            print_model_info(model)
+            print_model_version_info(mv)
+
```
Comment (BenWilson2) [inline mlflow/tracking/client.py:2966]: I think you need another line space here to make sphinx happy

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -2904,3 +2909,257 @@ def print_model_version_info(mv):
...
+            Version: 1
+            Aliases: []
+
+            --Model--
+            name: RandomForestRegression
+            aliases: {"test-alias": "1"}
+            --Model Version--
+            Name: RandomForestRegression
+            Version: 1
+            Aliases: ["test-alias"]
```
Comment (BenWilson2) [inline mlflow/tracking/client.py:2984]: You'll want a space ( or 2?) after the block quote here

**Review by BenWilson2 — APPROVED** [review]
LGTM! :)

**Pull request merged**